server/src/main/java/org/elasticsearch/search/aggregations/metrics/GeoBoundsAggregator.java

```diff
@@ -44,12 +45,7 @@
 
     private final ValuesSource.GeoPoint valuesSource;
     private final boolean wrapLongitude;
-    DoubleArray tops;
-    DoubleArray bottoms;
-    DoubleArray posLefts;
-    DoubleArray posRights;
-    DoubleArray negLefts;
-    DoubleArray negRights;
+    private ObjectArray<GeoExtent> extents;
```

Comment:
when using ObjectArray, what is the best way to update the size calculation to accurately circuit-break on the size of the ObjectArray given the size of GeoExtent

Reply:
This is a very good point. This pattern is used in other aggregations (e.g percentiles) and it seems nothing is done in respect the side of the objects.

Having a look into the [code](https://github.com/elastic/elasticsearch/blob/e63d52af63ed66bbc8400935efc4c359f84b5a08/server/src/main/java/org/elasticsearch/common/util/BigArrays.java#L747), it seems the objects size is not taking into account:

- When creating an [`ObjectArrayWrapper`](https://github.com/elastic/elasticsearch/blob/e63d52af63ed66bbc8400935efc4c359f84b5a08/server/src/main/java/org/elasticsearch/common/util/BigArrays.java#L326), the size calculated is the shallow size of the array.

- When creating an [`BigObjectArray`](https://github.com/elastic/elasticsearch/blob/e63d52af63ed66bbc8400935efc4c359f84b5a08/server/src/main/java/org/elasticsearch/common/util/BigObjectArray.java#L33), the size is calculated as if it were an integer array (this looks like a copy / paste bug, see javadocs from class).

Need to think / discuss what to do next.

Reply:
@polyfractal can we get your opinion here?

Reply:
Agreed this is an issue, I don't think any of the users of ObjectArrays accurately track the internal object's size... just the shallow size of the array itself.  Not sure I have a good solution.  

We could perhaps add an `objectSize` parameter to the constructor, which is used as a rough estimate of each object, and increment the breaker accordingly when we grow the array?

Reply:
I think that wis complex as this Object arrays are used in other aggregations where objects sizes are difficult to estimate. For the time being I have added code in the aggregation that will track it.

Note that for BigArrays, they are only tracked when the size of the arrays > page size (16kb)